src/components/RadioButton/RadioButton.scss

```diff
@@ -2,56 +2,121 @@
 
 .RadioButton {
   position: relative;
+  margin: var(--p-choice-margin, 0);
 }
 
 .Input {
-  margin: 0;
-  padding: 0;
-  opacity: 0;
+  @include visually-hidden;
 
-  &:focus,
-  &:active,
-  &:checked {
-    + .Backdrop {
-      background: color('indigo');
+  &:not(.globalTheming) {
+    &:focus,
+    &:active,
+    &:checked {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        background: color('indigo');
+      }
     }
-  }
 
-  &:checked {
-    ~ .Icon {
-      transform: translate(-50%, -50%) scale(1);
+    &:checked {
+      // stylelint-disable-next-line selector-max-specificity
+      ~ .Icon {
+        transform: translate(-50%, -50%) scale(1);
+      }
+    }
+
+    &:focus,
+    &:active {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop(focused);
+      }
     }
-  }
 
-  &:focus,
-  &:active {
-    + .Backdrop {
-      @include control-backdrop(focused);
+    &:disabled {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop(disabled);
+      }
+      // stylelint-disable-next-line selector-max-specificity
+      ~ .Icon {
+        background-color: color('sky', 'dark');
+      }
     }
   }
 
-  &:disabled {
-    + .Backdrop {
-      @include control-backdrop(disabled);
+  &.globalTheming {
+    &:focus {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include focus-ring-animated($show: 1);
```

Comment:
This API feels a bit strange to me. What about setting it to accept states, similar to how the `control-backdrop` mixin operates. That way you could pass `focused` as a state to indicate that it should be visible.